classes/ObjectModel.php

```diff
@@ -675,7 +675,15 @@ public function update($null_values = false)
                 // A little explanation of what we do here : we want to create multishop entry when update is called, but
                 // only if we are in a shop context (if we are in all context, we just want to update entries that alread exists)
                 $shop_exists = Db::getInstance()->getValue('SELECT '.$this->def['primary'].' FROM '._DB_PREFIX_.$this->def['table'].'_shop WHERE '.$where);
+
+                $display_multishop_checkboxes = Shop::isFeatureActive() && Shop::getContext() != Shop::CONTEXT_SHOP;
+
                 if ($shop_exists) {
+                    foreach($fields as $key => $val) {
+                        if(!array_key_exists($key, $this->update_fields) && $display_multishop_checkboxes) {
+                            unset($fields[$key]);
+                        }
+                    }
```

Comment:
I think there should be an `if ($display_multishop_checkboxes)` before the foreach, so that it won't loop over `$fields` needlessly when `$display_multishop_checkboxes` is false.

Also, please don't forget that according to PSR-2 there should be a space before the opening parenthesis on `foreach` and `if`.